addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanner.java

```diff
@@ -168,11 +181,25 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
             List<String> allowedWildcardSources =
                     getAllowedWildcardSources(unifiedPolicyText, origin);
             if (!allowedWildcardSources.isEmpty()) {
+                List<String> allowedDirectivesWithoutFallback =
+                        allowedWildcardSources.stream()
+                                .distinct()
+                                .filter(DIRECTIVES_WITHOUT_FALLBACK::contains)
+                                .collect(Collectors.toList());
                 String allowedWildcardSrcs =
                         allowedWildcardSources.toString().replace("[", "").replace("]", "");
                 String wildcardSrcDesc =
                         Constant.messages.getString(
                                 MESSAGE_PREFIX + "wildcard.desc", allowedWildcardSrcs);
+                if (!allowedDirectivesWithoutFallback.isEmpty()) {
+                    wildcardSrcDesc +=
+                            Constant.messages
+                                    .getString(
+                                            "pscanrules.cspscanner.desc.extended",
+                                            allowedDirectivesWithoutFallback.toString())
+                                    .replace("[", "")
+                                    .replace("]", "");
```

Comment:
`String.join(", ", allowedDirectivesWithoutFallback)`, `allowedWildcardSrcs` can also be tweaked.